File: java/app/src/main/java/com/pgssoft/mvvm/views/activities/RateActivity.java
package com.pgssoft.mvvm.views.activities;

import android.content.Context;
import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.pgssoft.mvvm.MVVMApplication;
import com.pgssoft.mvvm.R;
import com.pgssoft.mvvm.databinding.ActivityRateBinding;
import com.pgssoft.mvvm.services.ServiceProvider;
import com.pgssoft.mvvm.viewmodels.RateActivityViewModel;
import com.pgssoft.mvvm.viewmodels.interfaces.IRateActivityAccess;

/**
 * Created by bstokrocki on 31.01.2017.
 */
public class RateActivity extends BaseViewModelActivity<RateActivityViewModel>
        implements IRateActivityAccess {
    public static final String EXTRA_RATE_ID = "EXTRA_RATE_ID";

    private ServiceProvider serviceProvider;
    private ActivityRateBinding binding;
    private RateActivityViewModel viewModel;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setTitle(R.string.label_calculator);

        binding = DataBindingUtil.setContentView(this, R.layout.activity_rate);

        String rateId = getIntent().getStringExtra(EXTRA_RATE_ID);
        ServiceProvider serviceProvider = MVVMApplication.getServiceProvider();

        viewModel = new RateActivityViewModel(serviceProvider.getRepository(),
                serviceProvider.getSchedulerService(), rateId);

        binding.setViewModel(viewModel);
    }

    public static Intent prepareStartIntent (Context context, String rateCurrencyCode) {
        Intent intent = new Intent(context, RateActivity.class);
        intent.putExtra(EXTRA_RATE_ID, rateCurrencyCode);

        return intent;
    }
}

File: kotlin/app/src/main/java/com/pgssoft/kotlinplayground/view/activity/RateActivity.kt
package com.pgssoft.kotlinplayground.view.activity

import android.content.Context
import android.content.Intent
import android.databinding.DataBindingUtil
import android.os.Bundle
import com.pgssoft.kotlinplayground.KotlinApp
import com.pgssoft.kotlinplayground.R
import com.pgssoft.kotlinplayground.databinding.ActivityRateBinding
import com.pgssoft.kotlinplayground.viewmodel.RateActivityViewModel
import com.pgssoft.kotlinplayground.viewmodel.iface.IRateActivityAccess

/**
 * Created by bstokrocki on 31.01.2017.
 */
class RateActivity : BaseViewModelActivity<RateActivityViewModel>(), IRateActivityAccess {
    companion object {
        val EXTRA_RATE_ID = "EXTRA_RATE_ID"

        fun prepareStartIntent(context: Context, rateCurrencyCode: String): Intent {
            val intent = Intent(context, RateActivity::class.java)
            intent.putExtra(EXTRA_RATE_ID, rateCurrencyCode)

            return intent
        }
    }

    override lateinit var viewModel: RateActivityViewModel
    private lateinit var binding: ActivityRateBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setTitle(R.string.label_calculator)

        binding = DataBindingUtil.setContentView(this, R.layout.activity_rate)

        val rateId = intent.getStringExtra(EXTRA_RATE_ID)
        val serviceProvider = KotlinApp.serviceProvider
        viewModel = RateActivityViewModel(serviceProvider.scheduler, serviceProvider.repository, rateId)

        binding.viewModel = viewModel
    }
}
